[CI/Build] Add support for Nvidia RTX PRO 6000 Blackwell

ikasas opened the pull request:
**What this PR does / why we need it**:
To support NVIDIA Blackwell GPUs (e.g., NVIDIA RTX PRO 6000 Blackwell Server Edition) in Docker containers, consistent with vLLM’s capabilities.

Related PR in vLLM: https://github.com/vllm-project/vllm/pull/19794
Related comment: https://github.com/LMCache/LMCache/issues/860#issuecomment-3006438720

**Special notes for your reviewers**:

**If applicable**:

- [x] this PR contains user facing changes - docs added
- [ ] this PR contains unit tests

ikasas committed:
[CI/Build] Add SM120 to the Dockerfile

Signed-off-by: ikasas <Sasaki.Kazuki@da.MitsubishiElectric.co.jp>

DongDongJu reviewed (approved):
Hello @ikasas,
Thanks for catching this.
LGTM

ikasas commented:
@DongDongJu, thank you for the approval.

@hickeyma and @Shaoting-Feng, could you please take a look at this pull request when you have a moment?

hickeyma reviewed (changes requested):
Thanks @ikasas for pushing the change.

You will also need to update the build in https://github.com/LMCache/LMCache/blob/dev/pyproject.toml#L142-L151 accordingly. 

Do you mind also adding `10.3` as it provides support for `GB300` and `B300`?

hickeyma commented on docker/Dockerfile:
```diff
@@ -45,7 +45,7 @@ RUN --mount=type=cache,target=/root/.cache/pip \
 
 # CUDA arch list used by torch
-ARG torch_cuda_arch_list='7.0 7.5 8.0 8.6 8.9 9.0 10.0+PTX'
+ARG torch_cuda_arch_list='7.0 7.5 8.0 8.6 8.9 9.0 10.0 12.0'
```
Yes, it is: https://developer.nvidia.com/cuda/gpus

hickeyma commented on docker/README.md:
```diff
@@ -170,7 +170,7 @@ All Dockerfiles support the following build arguments:
 | `max_jobs` | `2` | Max parallel jobs for build |
 | `nvcc_threads` | `8` | Number of nvcc threads |
-| `torch_cuda_arch_list` | `7.0 7.5 8.0 8.6 8.9 9.0 10.0+PTX` | CUDA architectures |
+| `torch_cuda_arch_list` | `7.0 7.5 8.0 8.6 8.9 9.0 10.0 12.0` | CUDA architectures |
```
https://developer.nvidia.com/cuda/gpus

ikasas committed:
Add build support for Nvidia B300 and RTX PRO 6000 Blackwell

Signed-off-by: ikasas <Sasaki.Kazuki@da.MitsubishiElectric.co.jp>

hickeyma committed:
Merge branch 'dev' into add-sm120-dockerfile

ikasas commented:
@hickeyma, thank you for the quick review. I’ve updated `pyproject.toml` and added `10.3` to support NVIDIA GB300 and B300, as suggested. Could you please take another look?

ikasas commented:
@hickeyma, I apologize for the repeated messages. 

To compile for Compute Capability 10.3 (targeting NVIDIA B300 and GB300), CUDA version 12.9 appears to be required.

Since the current CUDA version in LMCache is 12.8, would it be acceptable to limit this change to supporting Compute Capability 12.0 (targeting NVIDIA RTX PRO 6000 Blackwell) only?

References:
https://docs.nvidia.com/cuda/archive/12.9.0/cuda-toolkit-release-notes/index.html#new-features
https://docs.nvidia.com/cuda/archive/12.8.0/cuda-toolkit-release-notes/index.html#new-features

valtab commented:
need support for [Nvidia RTX PRO 6000 Blackwell]

geraldstanje commented:
hi any update on this? cc @DongDongJu @hickeyma
